core/src/main/java/org/elasticsearch/index/query/MatchQueryBuilder.java

```diff
@@ -167,7 +183,12 @@ public String analyzer() {
         return this.analyzer;
     }
 
-    /** Sets a slop factor for phrase queries */
+    /**
+     * Sets a slop factor for phrase queries
+     *
+     * @deprecated for phrase queries use {@link MatchPhraseQueryBuilder}
+     */
+    @Deprecated
```

Comment:
Shall we also deprecate the Type enum in MatchQuery so we don't forget to remove it at some point?